jenkins/common.groovy

```diff
@@ -139,38 +172,12 @@ def build(String ref, dockerContainer, String assetSource, String envName, Boole
 
   withCredentials([usernamePassword(credentialsId:  "${drupalCred}", usernameVariable: 'DRUPAL_USERNAME', passwordVariable: 'DRUPAL_PASSWORD')]) {
     dockerContainer.inside(DOCKER_ARGS) {
-      def buildLog = "/application/${envName}-build.log"
-
-      try {
-	sh "cd /application && jenkins/build.sh --envName ${envName} --assetSource ${assetSource} --drupalAddress ${drupalAddress} ${drupalMode} --buildLog ${buildLog}"
-      } catch (error) {
-	// Look for broken links
-	def csvFileName = "${envName}-broken-links.csv" // For use within the docker container
-	def csvFile = "${WORKSPACE}/vets-website/${csvFileName}" // For use outside of the docker context
-
-	// Ensure the file isn't there if we had to rebuild
-	if (fileExists(csvFile)) {
-	  sh "rm /application/${csvFileName}"
-	}
-
-	// Output a csv file with the broken links
-	sh "cd /application && jenkins/glean-broken-links.sh ${buildLog} ${csvFileName}"
-	if (fileExists(csvFile)) {
-	  echo "Found broken links; notifying the Slack channel."
-	  // TODO: Move this slackUploadFile to cacheDrupalContent and update the echo statement above
-	  // slackUploadFile(filePath: csvFile, channel: 'dev_null', failOnError: true, initialComment: "Found broken links in the ${envName} build on `${env.BRANCH_NAME}`.")
-
-	  // Until slackUploadFile works...
-	  def linkCount = sh(returnStdout: true, script: "cd /application && wc -l ${csvFileName} | cut -d ' ' -f1") as Integer
-	  slackSend message: "@here ${linkCount - 1} broken links found in the ${envName} build on `${env.BRANCH_NAME}`\n${env.RUN_DISPLAY_URL}".stripMargin(),
-	    color: 'danger',
-	    failOnError: true,
-	    channel: 'cms-ci'
-	} else {
-	  echo "Did not find broken links."
-	}
-
-	throw error
+      def buildLogPath = "/application/${envName}-build.log"
+
+      sh "cd /application && jenkins/build.sh --envName ${envName} --assetSource ${assetSource} --drupalAddress ${drupalAddress} ${drupalMode} --buildLog ${buildLogPath}"
+
+      if (envName == 'vagovprod') {
+	checkForBrokenLinks(buildLogPath, envName)
```

Comment:
Is this executed for all branches? If there is a broken link, and all feature branches run this, won't we see duplicate Slack messages coming into content?

Reply:
Another imperfect solution. Yes. :disappointed: 

I'm not sure if there's any kind of throttling we can do here. It's not ideal for sure.

On the other hand, the constant bombarding of broken link notifications may cause them to get fixed more quickly? Maybe they'll be annoying enough to give pause before introducing broken links in the first place?

Definitely not how I want to do it, but I don't know how to notify only as often as it's needed. If you or anybody else has ideas, I'm all ears! (Or eyes? :thinking:)

Reply:
If you think this is going to be too much, I can remove the `@here` so it's less spammy.

Reply:
What I'm afraid is going to happen is that it'll be so spammy that engineers will have to scatter to fix the link 😬

Reply:
As far as I know, we can't share the state between branches. That is, when one branch finds broken links, check to see if the slack channel has been notified of these same broken links recently before sending out a new notification. That gets pretty hairy.

I think even if we could decouple this check from feature branch pushes (which trigger builds), and run it on like a cron job (which, again, is hairy for a lot of reasons, I think), we'd still _want_ to check each branch build to make sure the broken links aren't introduced by a `vets-website` change.

I will emphasize again that this is MVP. It could be spammy, true, but it could also be that this sort of early warning system will be enough. The channel is notified more-or-less as soon as a broken prod link is detected.